### DISCUSSION TITLE
Docs improvement on generating clients

### DISCUSSION THREAD

**Discussion by urosjarc** [fastapi/fastapi#5616] [Questions] [answered]
npx @openapitools/openapi-generator-cli generate \
   -i http://localhost:8000/openapi.json \
   -g typescript-fetch \
   -o src/services

https://www.npmjs.com/package/@openapitools/openapi-generator-cli

**Comment by tiangolo** [comment, marked as answer]
Could you elaborate, please?

**Reply by urosjarc** [reply]
Joust want to tell that there are official openapi way to generate client sdk for any programming language... https://openapi-generator.tech/docs/generators/

Beginners coming on your site should know about that this is possible...
The downsides of this lib is that needs java jdk :(

Also comparing openapi-typescript-codegen with [@openapitools/openapi-generator-cli](https://www.npmjs.com/package/@openapitools/openapi-generator-cli) shows that openapi-typescript-codegen has
better code structure... They use static methods while generator-cli uses classes that you have to construct first.

**Reply by tiangolo** [reply]
Hmm, and did you check the existing docs? https://fastapi.tiangolo.com/advanced/generate-clients/

**Reply by urosjarc** [reply]
You are right it slipped through my sight https://fastapi.tiangolo.com/advanced/generate-clients/#openapi-client-generators

One more thing...
It would be cool if you could fix /openapi.json route to return custom json...
I have tried to override it but it does not want to register custom route...

I end up making my own route to return changed configuration...

```python

@app.get('/schema', include_in_schema=False)
def openapi():
	return utils.openapi(app.openapi())

```

**Reply by tiangolo** [reply]
Probably this section in the docs? https://fastapi.tiangolo.com/advanced/extending-openapi/

**Reply by urosjarc** [reply]
GODDAMMIT this project have no flaws! I didn't see it.

 I'm sorry i'm used to having read shity documentation for xY npm package.

**Reply by tiangolo** [reply]
Hehe, cool! Thanks!
